src/cunumeric/matrix/solve.cc

```diff
@@ -0,0 +1,115 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/matrix/solve.h"
+#include "cunumeric/matrix/solve_template.inl"
+
+#include <cblas.h>
+#include <lapack.h>
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <>
+struct SolveImplBody<VariantKind::CPU, LegateTypeCode::FLOAT_LT> {
+  void operator()(int32_t m, int32_t n, int32_t nrhs, float* a, float* b)
+  {
+    const char trans = 'N';
+    int32_t info     = 0;
+
+    auto kind = CuNumeric::has_numamem ? Memory::Kind::SOCKET_MEM : Memory::Kind::SYSTEM_MEM;
```

Comment:
I assume the reason to consider SOCKET_MEM here is that we expect the user will allocate most of the available memory to SOCKET_MEM vs SYSTEM_MEM (if they allocate any SOCKET_MEM at all). Should we be doing this in other CPU tasks as well?

Reply:
Almog (the original author of this PR)  wrote this so I didn't pay close attention to it. I think always using `SYSTEM_MEM` should be okay as well, because if there's a socket memory, OpenMP processors would be also available and preferred. I don't think we need to change this code nor do we need to go fix other CPU tasks to make them consistent.